src/levelset/levelSetSegmentationObject.cpp

```diff
@@ -331,6 +331,671 @@ void LevelSetSegmentationSurfaceInfo::evalProjectionOnVertex(const std::array<do
     (*projectionNormal) = {0., 0., 0.};
 }
 
+/*!
+ * Evaluate the projection of the given point on the surface created based on
+ * the points representing the specified segment. The surface passes from these
+ * points and is verical to the normal vectors associated with them.
+ *
+ * In case the segment is a line, the surface degenerates to a curve.
+ *
+ * @param[in] point are the coordinates of the given point
+ * @param[in] segmentItr is an iterator pointing to the segment on which the surface
+ * will be created.
+ * @param[out] projectionPoint The coordinates of the projection point on the surface.
+ * @param[out] projectionNormal The coordinates of the norrmal to the surface vector on the surface
+ * projection point.
+ */
+void LevelSetSegmentationSurfaceInfo::evalHighOrderProjectionOnLine(const std::array<double,3> &point,
+                                                                    const SegmentConstIterator &segmentItr,
+                                                                    std::array<double, 3> *projectionPoint,
+                                                                    std::array<double, 3> *projectionNormal) const
+{
+    // Get segment
+    const Cell &segment = *segmentItr;
+
+    // Get segment vertices
+    ConstProxyVector<long> segmentVertexIds = segment.getVertexIds();
+
+    std::array<double,3> point0 = m_surface->getVertexCoords(segmentVertexIds[0]);
+    std::array<double,3> point1 = m_surface->getVertexCoords(segmentVertexIds[1]);
+
+    // Get projection point on segment
+    int nSegmentVertices = segment.getVertexCount();
+    BITPIT_CREATE_WORKSPACE(t, double, nSegmentVertices, ReferenceElementInfo::MAX_ELEM_VERTICES);
```

Comment:
This code is specific for the line element, ```t``` can be declared as an array of size ReferenceElementInfo::MAX_ELEM_VERTICES.

Reply:
As a side note, we should modify ReferenceElementInfo to allow accessing information on reference elements at compile time:
```
	const ReferenceElementInfo &referenceSegment = ReferenceElementInfo::getInfo(ElementType::LINE);
	std::array<double, referenceSegment.nVertices> t;
```

Reply:
I deleted the usage of BITPIT_CREATE_WORKSPACE for lines and triangles.